cpu/atmega_common/periph/rtt.c

```diff
@@ -0,0 +1,288 @@
+    _asynch_wait();
+
+    return (((uint32_t)rtt_state.ext_cnt << 8) | (uint32_t)TCNT2);
+}
+
+void rtt_set_counter(uint32_t counter)
+{
+    /* Wait until not busy anymore (should be immediate) */
+    DEBUG("RTT sleeps until safe to write TCNT2\n");
+    _asynch_wait();
+
+    rtt_state.ext_cnt = (uint16_t)(counter >> 8);
+    TCNT2 = (uint8_t)counter;
+
+    DEBUG("RTT set counter TCNT2: %" PRIu8 ", ext_cnt: %" PRIu16 "\n",
+          TCNT2, rtt_state.ext_cnt);
+}
+
+void rtt_set_alarm(uint32_t alarm, rtt_cb_t cb, void *arg)
+{
+    uint32_t now;
+
+    /* Disable alarm interrupt */
+    TIMSK2 &= ~(1 << OCIE2A);
+    rtt_state.alarm_cb = NULL;
+
+    /* Make sure it is safe to read TCNT2, in case we just woke up, and */
+    /* safe to write OCR2B (in case it was busy) */
+    DEBUG("RTC sleeps until safe read TCNT2 and to write OCR2B\n");
+    TCCR2A = 0;
+    _asynch_wait();
+
+    now = ((uint32_t)rtt_state.ext_cnt << 8) | TCNT2;
+
+    if (alarm < now) {
+        DEBUG("RTC alarm set in the past. Time: %" PRIu32 " seconds, alarm: %"
+              PRIu32 "\n", now, alarm);
+        cb(arg);
+        return;
+    }
+
+    /* Set the alarm value */
+    rtt_state.ext_comp = (uint16_t)(alarm >> 8);
+    OCR2A = (uint8_t)alarm;
+
+    DEBUG("RTT set alarm TCNT2: %" PRIu8 ", OCR2A: %" PRIu8 "\n", TCNT2, OCR2A);
+
+    /* Interrupt safe order of assignment */
+    rtt_state.alarm_arg = arg;
+    rtt_state.alarm_cb = cb;
+
+    /* Enable alarm interrupt only if it will trigger before overflow */
+    if (rtt_state.ext_comp <= rtt_state.ext_cnt) {
+        /* Clear interrupt flag */
+        TIFR2 = (1 << OCF2A);
+
+        /* Enable interrupt */
+        TIMSK2 |= (1 << OCIE2A);
+
+        DEBUG("RTT alarm interrupt active\n");
+    }
+    else {
+        DEBUG("RTT alarm interrupt not active\n");
+    }
+}
+
+uint32_t rtt_get_alarm(void)
+{
+    return (((uint32_t)rtt_state.ext_comp << 8) | (uint32_t)OCR2A);
+}
+
+void rtt_clear_alarm(void)
+{
+    /* Disable alarm interrupt */
+    TIMSK2 &= ~(1 << OCIE2A);
+
+    /* Clear interrupt flag */
+    TIFR2 = (1 << OCF2A);
+
+    /* Interrupt safe order of assignment */
+    rtt_state.alarm_cb = NULL;
+    rtt_state.alarm_arg = NULL;
+}
+
+void rtt_poweron(void)
+{
+    power_timer2_enable();
+}
+
+void rtt_poweroff(void)
+{
+    power_timer2_disable();
+}
+
+void _asynch_wait(void)
+{
+    /* Wait until all busy flags clear. According to the datasheet,
+     * this can take up to 2 positive edges of TOSC1 (32kHz). */
+    while (ASSR & ((1 << TCN2UB) | (1 << OCR2AUB) | (1 << OCR2BUB)
+                   | (1 << TCR2AUB) | (1 << TCR2BUB))) ;
+}
+
+ISR(TIMER2_OVF_vect) {
+    __enter_isr();
+
+    rtt_state.ext_cnt++;
+
+    /* Enable RTT alarm if overflowed enough times */
+    if (rtt_state.ext_comp <= rtt_state.ext_cnt) {
+        /* Clear interrupt flag */
+        TIFR2 = (1 << OCF2A);
+
+        /* Enable interrupt */
+        TIMSK2 |= (1 << OCIE2A);
+    }
+
+#if MODULE_PERIPH_RTC
```

Comment:
While this seems to work, I would prefer `#ifdef MODULE_PERIPH_RTC`. This would be consistent to the negation of the macro, as

``` C
#ifndef MODULE_PERIPH_RTC
```
will work but
```C
#if !MODULE_PERIPH_RTC
```
will not work. (See this as a suggestion, I will not block the PR on this.)